server/fake_require.js

```diff
@@ -65,7 +68,10 @@ module.exports = {
       
       // Remove new cache entries.
       let newDeps = _.difference(Object.keys(require.cache), loadedDeps);
-      newDeps.forEach((k) => delete require.cache[k]);
+      newDeps.forEach(k => delete require.cache[k]);
+
+      // Also, remove these entries from MY module's children list.
+      module.children = module.children.filter(c => originalChildren.indexOf(c.id) != -1);
```

Comment:
Should this just be ```.indexOf(c)``` rather than ```.indexOf(c.id)```?

Also, since you already have underscore, is it more readable to use 

```
_.contains(originalChildren, c)
```